src/api/carts/query-keys.ts

```diff
@@ -0,0 +1,6 @@
+import { createQueryKeys } from '@lukemorales/query-key-factory';
+
+export const cartKeys = createQueryKeys('carts', {
+  list: (filters) => ['carts', filters],
+  detail: (id) => [id],
```

Comment:
Why do you need to pass the original query key in `list` but not in `detail`? If we create a query key named `carts`, how it will be for `list` and `details`? 

list: ['carts', 'carts', filters] (you're passing carts on `list` too)
detail: ['carts', id]

or

list: ['carts', filters]
detail: [id]

Reply:
You're right, there is no need to pass the original query key name